From 86826f5e5927aff2b3dc73818f5f4a172119387e Mon Sep 17 00:00:00 2001
From: Xavier Hanin <xavier@apache.org>
Date: Thu, 8 Mar 2007 09:17:54 +0000
Subject: [PATCH] FIX: bad tests for conflict managers use the ivy default
 cache instead of a fresh one FIX: set last modified from artifact resource
 when generating default ivy file (related to IVY-407)

git-svn-id: https://svn.apache.org/repos/asf/incubator/ivy/core/trunk@515980 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/ivy/core/resolve/IvyNode.java     |  1 +
 .../apache/ivy/plugins/resolver/BasicResolver.java    |  4 ++++
 .../plugins/conflict/LatestConflictManagerTest.java   |  8 ++++++++
 .../plugins/conflict/RegexpConflictManagerTest.java   | 10 ++++++++++
 .../plugins/conflict/StrictConflictManagerTest.java   | 11 +++++++++++
 .../conflict/ivyconf-latest-time-transitivity.xml     |  2 +-
 .../ivy/plugins/conflict/ivyconf-latest-time.xml      |  2 +-
 .../apache/ivy/plugins/conflict/ivyconf-latest.xml    |  2 +-
 .../ivy/plugins/conflict/ivyconf-regexp-test.xml      |  3 ++-
 .../ivy/plugins/conflict/ivyconf-strict-test.xml      |  3 ++-
 10 files changed, 41 insertions(+), 5 deletions(-)

diff --git a/src/java/org/apache/ivy/core/resolve/IvyNode.java b/src/java/org/apache/ivy/core/resolve/IvyNode.java
index 8fa19e07b..23ba2f114 100644
--- a/src/java/org/apache/ivy/core/resolve/IvyNode.java
+++ b/src/java/org/apache/ivy/core/resolve/IvyNode.java
@@ -20,6 +20,7 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
diff --git a/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java b/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
index 30a959c70..1f2cee3b5 100644
--- a/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
+++ b/src/java/org/apache/ivy/plugins/resolver/BasicResolver.java
@@ -221,6 +221,10 @@ public ResolvedModuleRevision getDependency(DependencyDescriptor dd, ResolveData
                 }
                 return null;
             } else {
+            	long lastModified = artifactRef.getLastModified();
+            	if (lastModified != 0 && md instanceof DefaultModuleDescriptor) {
+            		((DefaultModuleDescriptor) md).setLastModified(lastModified);
+            	}
                 Message.verbose("\t"+getName()+": no ivy file found for "+mrid+": using default data");            
                 logIvyNotFound(mrid);
     	        if (isDynamic) {
diff --git a/test/java/org/apache/ivy/plugins/conflict/LatestConflictManagerTest.java b/test/java/org/apache/ivy/plugins/conflict/LatestConflictManagerTest.java
index d7ec5cde0..bbdeb6ed8 100644
--- a/test/java/org/apache/ivy/plugins/conflict/LatestConflictManagerTest.java
+++ b/test/java/org/apache/ivy/plugins/conflict/LatestConflictManagerTest.java
@@ -30,15 +30,23 @@
 import org.apache.ivy.core.report.ResolveReport;
 import org.apache.ivy.core.resolve.IvyNode;
 import org.apache.ivy.core.resolve.ResolveOptions;
+import org.apache.ivy.util.FileUtil;
 
 public class LatestConflictManagerTest extends TestCase {
 
 	private Ivy ivy;
+    private File _cache;
 
 	protected void setUp() throws Exception {
 		ivy = new Ivy();
 		ivy.configure(LatestConflictManagerTest.class
 				.getResource("ivyconf-latest.xml"));
+        _cache = new File("build/cache");
+        _cache.mkdirs();
+	}
+	
+	protected void tearDown() throws Exception {
+		FileUtil.forceDelete(_cache);
 	}
 
 	// Test case for issue IVY-388
diff --git a/test/java/org/apache/ivy/plugins/conflict/RegexpConflictManagerTest.java b/test/java/org/apache/ivy/plugins/conflict/RegexpConflictManagerTest.java
index f7ef89fc3..5189a4c30 100644
--- a/test/java/org/apache/ivy/plugins/conflict/RegexpConflictManagerTest.java
+++ b/test/java/org/apache/ivy/plugins/conflict/RegexpConflictManagerTest.java
@@ -21,23 +21,33 @@
  * @author Anders janmyr
  */
 
+import java.io.File;
+
 import junit.framework.TestCase;
 
 import org.apache.ivy.Ivy;
 import org.apache.ivy.core.cache.CacheManager;
 import org.apache.ivy.core.resolve.ResolveOptions;
+import org.apache.ivy.util.FileUtil;
 
 
 public class RegexpConflictManagerTest extends TestCase
 {
     private Ivy ivy;
+    private File _cache;
 
     protected void setUp() throws Exception
     {
         ivy = new Ivy();
         ivy.configure( RegexpConflictManagerTest.class
                 .getResource( "ivyconf-regexp-test.xml" ) );
+        _cache = new File("build/cache");
+        _cache.mkdirs();
     }
+	
+	protected void tearDown() throws Exception {
+		FileUtil.forceDelete(_cache);
+	}
 
     public void testNoApiConflictResolve() throws Exception
     {
diff --git a/test/java/org/apache/ivy/plugins/conflict/StrictConflictManagerTest.java b/test/java/org/apache/ivy/plugins/conflict/StrictConflictManagerTest.java
index 968481f1c..5ef4504eb 100644
--- a/test/java/org/apache/ivy/plugins/conflict/StrictConflictManagerTest.java
+++ b/test/java/org/apache/ivy/plugins/conflict/StrictConflictManagerTest.java
@@ -17,18 +17,29 @@
  */
 package org.apache.ivy.plugins.conflict;
 
+import java.io.File;
+
 import junit.framework.TestCase;
 
 import org.apache.ivy.Ivy;
 import org.apache.ivy.core.cache.CacheManager;
 import org.apache.ivy.core.resolve.ResolveOptions;
+import org.apache.ivy.util.FileUtil;
 
 public class StrictConflictManagerTest extends TestCase {
 	private Ivy ivy;
+    private File _cache;
+
     protected void setUp() throws Exception {
         ivy = new Ivy();
         ivy.configure(StrictConflictManagerTest.class.getResource("ivyconf-strict-test.xml"));
+        _cache = new File("build/cache");
+        _cache.mkdirs();
     }
+	
+	protected void tearDown() throws Exception {
+		FileUtil.forceDelete(_cache);
+	}
 
     public void testInitFromConf() throws Exception {
         ConflictManager cm = ivy.getSettings().getDefaultConflictManager();
diff --git a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time-transitivity.xml b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time-transitivity.xml
index 3ab29c5fe..59906826c 100644
--- a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time-transitivity.xml
+++ b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time-transitivity.xml
@@ -1,6 +1,6 @@
 <ivyconf>
 
-    <conf defaultResolver="test" defaultConflictManager="latest-time" />
+    <conf defaultCache="build/cache" defaultResolver="test" defaultConflictManager="latest-time" />
 
     <resolvers>
        <filesystem name="test" latest="latest-time" checkmodified="true">
diff --git a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time.xml b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time.xml
index b2dd5eb75..a5aa68425 100644
--- a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time.xml
+++ b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest-time.xml
@@ -1,6 +1,6 @@
 <ivyconf>
 
-    <conf defaultResolver="test" defaultConflictManager="latest-time" />
+    <conf defaultCache="build/cache" defaultResolver="test" defaultConflictManager="latest-time" />
 
     <resolvers>
         <filesystem name="test">
diff --git a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest.xml b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest.xml
index d872ff97e..87860d2e7 100644
--- a/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest.xml
+++ b/test/java/org/apache/ivy/plugins/conflict/ivyconf-latest.xml
@@ -1,6 +1,6 @@
 <ivyconf>
 
-    <conf defaultResolver="test"/>
+    <conf defaultCache="build/cache" defaultResolver="test"/>
 
     <resolvers>
         <filesystem name="test">
diff --git a/test/java/org/apache/ivy/plugins/conflict/ivyconf-regexp-test.xml b/test/java/org/apache/ivy/plugins/conflict/ivyconf-regexp-test.xml
index 7f5b65c42..642a5c215 100644
--- a/test/java/org/apache/ivy/plugins/conflict/ivyconf-regexp-test.xml
+++ b/test/java/org/apache/ivy/plugins/conflict/ivyconf-regexp-test.xml
@@ -1,6 +1,7 @@
 <ivyconf>
 
-    <conf defaultResolver="test"
+    <conf defaultCache="build/cache"
+          defaultResolver="test"
           defaultConflictManager="regexp"
             />
 
diff --git a/test/java/org/apache/ivy/plugins/conflict/ivyconf-strict-test.xml b/test/java/org/apache/ivy/plugins/conflict/ivyconf-strict-test.xml
index 189a405fa..291ba68c3 100644
--- a/test/java/org/apache/ivy/plugins/conflict/ivyconf-strict-test.xml
+++ b/test/java/org/apache/ivy/plugins/conflict/ivyconf-strict-test.xml
@@ -1,6 +1,7 @@
 <ivyconf>
 
-  <conf defaultResolver="test"
+  <conf defaultCache="build/cache"
+        defaultResolver="test"
         defaultConflictManager="strict"
   />
   
